File: core/src/main/java/com/zeyad/rxredux/core/redux/StateReducer.java
package com.zeyad.rxredux.core.redux;

import io.reactivex.annotations.NonNull;

/** @author by Zeyad. */
public interface StateReducer<S> {
    @NonNull
    S reduce(@NonNull Object newResult, @NonNull String event, @NonNull S currentStateBundle);
}

File: core/src/main/java/com/zeyad/rxredux/core/StateReducer.kt
package com.zeyad.rxredux.core

import io.reactivex.annotations.NonNull

/**
 * @author ZIaDo on 2/27/18.
 */
interface StateReducer<S> {
    @NonNull
    fun reduce(@NonNull newResult: Any?, @NonNull event: String?, @NonNull currentStateBundle: S?): S
}